std/bitmanip.d

```diff
@@ -4005,3 +4005,444 @@ auto bitsSet(T)(T value) @nogc pure nothrow
     foreach (i; 0..63)
         assert(bitsSet(1UL << i).equal([i]));
 }
+
+pure @safe @nogc nothrow {
+
+private template upTo(T)
+{
+    enum upTo = T.sizeof * 8UL;
+}
+
+private template minUnsignedIntegral(T)
+{
+    static if (is(T == byte))
+    {
+        alias minUnsignedIntegral = ubyte;
+    }
+    else static if (is(T == short))
+    {
+        alias minUnsignedIntegral = ushort;
+    }
+    else static if (is(T == int))
+    {
+        alias minUnsignedIntegral = uint;
+    }
+    else static if (is(T == long))
+    {
+        alias minUnsignedIntegral = ulong;
+    }
+    else
+    {
+        alias minUnsignedIntegral = T;
+    }
+}
+
+/** Tests if the bit `idx` is set in integer `bitfield`.
+If `idx >= T.sizeof * 8` the bahaviour is undefined.
+
+Params:
+   bitfield = the integer containing bit to test
+   idx = the index of the bit to test
+
+Returns: `true` if the bit is set, `false` otherwise.
+*/
+bool testBit(T)(const T bitfield, const ulong idx) if (isIntegral!T)
+{
```

Comment:
well your bitfield could be a `long` and then you accidentally change it to `int` - I think it's always a good style to check for invalid input ;-)